src/main/kotlin/org/lunakoly/quicklink/actions/CopyLineLinkAction.kt

```diff
@@ -52,18 +54,18 @@ class CopyLineLinkAction : DumbAwareAction() {
 
         val isSelection = editor.selectionModel.hasSelection()
 
-        val lineNumber: Pair<Int, Int>
-        val columnNumber: Pair<Int, Int>
+        val selection: Selection
         if (isSelection) {
             val startLine = 1 + editor.selectionModel.selectionStartPosition!!.getLine()
             val endLine = 1 + editor.selectionModel.selectionEndPosition!!.getLine()
             val startColumn = 1 + editor.selectionModel.selectionStartPosition!!.getColumn()
             val endColumn = 1 + editor.selectionModel.selectionEndPosition!!.getColumn()
-            lineNumber = Pair(startLine, endLine)
-            columnNumber = Pair(startColumn, endColumn)
+            val startOffset = LineOffset(startLine, startColumn)
+            val endOffset = LineOffset(endLine, endColumn)
+            selection = Selection.MultilineSelection(startOffset, endOffset)
         } else {
-            lineNumber = Pair(1 + editor.document.getLineNumber(editor.caretModel.offset), 0)
-            columnNumber = Pair(0, 0)
+            val lineOffset = LineOffset(1 + editor.document.getLineNumber(editor.caretModel.offset), 0)
+            selection = Selection.SingleLinkSelection(lineOffset)
         }
```

Comment:
In Kotlin it's possible to write `val selection = if (...) { ... } else { ... }` (or similarly with `when`), so I think it would be better to do so here

Reply:
Done.